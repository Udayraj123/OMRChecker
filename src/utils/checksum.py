import contextlib
import hashlib
from pathlib import Path


def calculate_file_checksum(file_path: Path | str, algorithm: str = "sha256") -> str:
    """Calculate the checksum of a file using the specified hashing algorithm.

    Args:
        file_path: Path to the file
        algorithm: Hash algorithm to use (md5, sha1, sha256, sha512)

    Returns:
        Hexadecimal string representation of the file's checksum

    Raises:
        FileNotFoundError: If the file doesn't exist
        ValueError: If the algorithm is not supported
    """
    file_path = Path(file_path)

    if not file_path.exists():
        msg = f"File not found: {file_path}"
        raise FileNotFoundError(msg)

    # Validate algorithm
    try:
        hasher = hashlib.new(algorithm)
    except ValueError as e:
        msg = f"Unsupported hash algorithm: {algorithm}"
        raise ValueError(msg) from e

    # Read file in chunks to handle large files efficiently
    with file_path.open("rb") as f:
        while chunk := f.read(8192):
            hasher.update(chunk)

    return hasher.hexdigest()


def print_file_checksum(file_path: Path | str, algorithm: str = "md5") -> None:
    """Calculate and print the checksum of a file.

    Args:
        file_path: Path to the file
        algorithm: Hash algorithm to use (md5, sha1, sha256, sha512)
    """
    with contextlib.suppress(FileNotFoundError, ValueError):
        calculate_file_checksum(file_path, algorithm)
