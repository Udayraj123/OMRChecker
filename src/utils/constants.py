import matplotlib
import numpy as np
from dotmap import DotMap

# Filenames
TEMPLATE_FILENAME = "template.json"
EVALUATION_FILENAME = "evaluation.json"
CONFIG_FILENAME = "config.json"

SUPPORTED_PROCESSOR_NAMES = [
    "CropOnMarkers",
    "CropPage",
    "FeatureBasedAlignment",
    "GaussianBlur",
    "Levels",
    "MedianBlur",
    "AutoRotate",
]

FIELD_LABEL_NUMBER_REGEX = r"([^\d]+)(\d*)"
#
ERROR_CODES = DotMap(
    {
        "MULTI_BUBBLE_WARN": 1,
        "NO_MARKER_ERR": 2,
    },
    _dynamic=False,
)

BUILTIN_BUBBLE_FIELD_TYPES = {
    "QTYPE_INT": {
        "bubbleValues": ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"],
        "direction": "vertical",
    },
    "QTYPE_INT_FROM_1": {
        "bubbleValues": ["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"],
        "direction": "vertical",
    },
    "QTYPE_MCQ4": {"bubbleValues": ["A", "B", "C", "D"], "direction": "horizontal"},
    "QTYPE_MCQ5": {
        "bubbleValues": ["A", "B", "C", "D", "E"],
        "direction": "horizontal",
    },
    # Note: you can put custom field types in template.json in the "customBubbleFieldTypes" key.
}

CUSTOM_BUBBLE_FIELD_TYPE_PATTERN = "^CUSTOM_.*$"

# TODO: Move TEXT_SIZE, etc into a class
TEXT_SIZE = 0.95
CLR_BLACK = (0, 0, 0)
CLR_DARK_GRAY = (100, 100, 100)
CLR_DARK_BLUE = (255, 20, 20)
CLR_DARK_GREEN = (20, 255, 20)
CLR_DARK_RED = (20, 20, 255)
CLR_NEAR_BLACK = (20, 20, 10)
CLR_GRAY = (130, 130, 130)
CLR_LIGHT_GRAY = (200, 200, 200)
CLR_GREEN = (100, 200, 100)
CLR_WHITE = (255, 255, 255)
MARKED_TEMPLATE_TRANSPARENCY = 0.65

hsv_white_low, hsv_white_high = np.array([0, 0, 100]), np.array([180, 150, 255])

MATPLOTLIB_COLORS = matplotlib.colors.get_named_colors_mapping()

WAIT_KEYS = DotMap(
    {
        "ENTER": 13,
        "ESCAPE": 27,
        "SPACE": 32,
    },
    _dynamic=False,
)

ZERO_MARGINS = {"top": 0, "bottom": 0, "left": 0, "right": 0}
