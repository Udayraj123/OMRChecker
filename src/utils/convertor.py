import os
import argparse
from PIL import Image
from pdf2image import convert_from_path

def convert_image(input_path, output_path, output_format):
    with Image.open(input_path) as img:
        if output_format == 'JPG':
            output_format = 'JPEG'
        img.save(output_path, output_format)

def convert_pdf_to_jpg(input_path, output_dir):
    pages = convert_from_path(input_path)
    for i, page in enumerate(pages):
        output_path = os.path.join(output_dir, f"page_{i + 1}.jpg")
        page.save(output_path, 'JPEG')

def bulk_convert(input_dir, output_dir, output_format):
    os.makedirs(output_dir, exist_ok=True)
    
    for root, _, files in os.walk(input_dir):
        for filename in files:
            input_path = os.path.join(root, filename)
            relative_path = os.path.relpath(root, input_dir)
            output_subdir = os.path.join(output_dir, relative_path)
            os.makedirs(output_subdir, exist_ok=True)

            if filename.lower().endswith(('.png', '.jpg', '.jpeg', '.tiff', '.tif')):
                name, _ = os.path.splitext(filename)
                output_path = os.path.join(output_subdir, f"{name}.{output_format.lower()}")
                convert_image(input_path, output_path, output_format)
                print(f"Converted {filename} to {output_format}")
            elif filename.lower().endswith('.pdf'):
                pdf_output_dir = os.path.join(output_subdir, os.path.splitext(filename)[0])
                os.makedirs(pdf_output_dir, exist_ok=True)
                convert_pdf_to_jpg(input_path, pdf_output_dir)
                print(f"Converted {filename} to JPG")
            else:
                print(f"Skipping unsupported file: {filename}")

def main():
    parser = argparse.ArgumentParser(description="Bulk image and PDF converter")
    parser.add_argument("input_dir", help="Input directory containing images and PDFs")
    parser.add_argument("output_dir", help="Output directory for converted files")
    parser.add_argument(
        "--format", 
        choices=['jpg', 'png', 'jpeg'], 
        default='jpg', 
        help="Output format for images (default: jpg)"
    )
    
    args = parser.parse_args()
    
    bulk_convert(args.input_dir, args.output_dir, args.format.upper())

if __name__ == "__main__":
    main()
