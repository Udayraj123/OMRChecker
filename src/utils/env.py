import os


class Env:
    # Container for all environment variables used in OMRChecker
    CI = os.environ.get("CI")
    OMR_CHECKER_CONTAINER = os.environ.get("OMR_CHECKER_CONTAINER", CI)
    VIRTUAL_ENV = os.environ.get("VIRTUAL_ENV", "Not Present")


env = Env()
