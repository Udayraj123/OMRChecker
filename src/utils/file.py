import argparse
import json
import os
from csv import QUOTE_NONNUMERIC
from time import localtime, strftime

import pandas as pd

from src.logger import logger


def load_json(path, **rest):
    try:
        with open(path, "r") as f:
            loaded = json.load(f, **rest)
    except json.decoder.JSONDecodeError as error:
        logger.critical(f"Error when loading json file at: '{path}'\n{error}")
        exit(1)
    return loaded


class Paths:
    def __init__(self, output_dir):
        self.output_dir = output_dir
        self.save_marked_dir = output_dir.joinpath("CheckedOMRs")
        self.results_dir = output_dir.joinpath("Results")
        self.manual_dir = output_dir.joinpath("Manual")
        self.errors_dir = self.manual_dir.joinpath("ErrorFiles")
        self.multi_marked_dir = self.manual_dir.joinpath("MultiMarkedFiles")


def setup_dirs_for_paths(paths):
    logger.info("Checking Directories...")
    for save_output_dir in [paths.save_marked_dir]:
        if not os.path.exists(save_output_dir):
            logger.info(f"Created : {save_output_dir}")
            os.makedirs(save_output_dir)
            os.mkdir(save_output_dir.joinpath("stack"))
            os.mkdir(save_output_dir.joinpath("_MULTI_"))
            os.mkdir(save_output_dir.joinpath("_MULTI_", "stack"))

    for save_output_dir in [paths.manual_dir, paths.results_dir]:
        if not os.path.exists(save_output_dir):
            logger.info(f"Created : {save_output_dir}")
            os.makedirs(save_output_dir)

    for save_output_dir in [paths.multi_marked_dir, paths.errors_dir]:
        if not os.path.exists(save_output_dir):
            logger.info(f"Created : {save_output_dir}")
            os.makedirs(save_output_dir)


def setup_outputs_for_template(paths, template):
    # TODO: consider moving this into a class instance
    ns = argparse.Namespace()
    logger.info("Checking Files...")

    # Include current output paths
    ns.paths = paths

    ns.empty_resp = [""] * len(template.output_columns)
    ns.sheetCols = [
        "file_id",
        "input_path",
        "output_path",
        "score",
    ] + template.output_columns
    ns.OUTPUT_SET = []
    ns.files_obj = {}
    TIME_NOW_HRS = strftime("%I%p", localtime())
    ns.filesMap = {
        "Results": os.path.join(paths.results_dir, f"Results_{TIME_NOW_HRS}.csv"),
        "MultiMarked": os.path.join(paths.manual_dir, "MultiMarkedFiles.csv"),
        "Errors": os.path.join(paths.manual_dir, "ErrorFiles.csv"),
    }

    for file_key, file_name in ns.filesMap.items():
        if not os.path.exists(file_name):
            logger.info(f"Created new file: '{file_name}'")
            # moved handling of files to pandas csv writer
            ns.files_obj[file_key] = file_name
            # Create Header Columns
            pd.DataFrame([ns.sheetCols], dtype=str).to_csv(
                ns.files_obj[file_key],
                mode="a",
                quoting=QUOTE_NONNUMERIC,
                header=False,
                index=False,
            )
        else:
            logger.info(f"Present : appending to '{file_name}'")
            ns.files_obj[file_key] = open(file_name, "a")

    return ns
