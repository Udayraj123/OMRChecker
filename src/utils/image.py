"""

 OMRChecker

 Author: Udayraj Deshmukh
 Github: https://github.com/Udayraj123

"""
import cv2
import matplotlib.pyplot as plt
import numpy as np

from src.logger import logger

plt.rcParams["figure.figsize"] = (10.0, 8.0)
CLAHE_HELPER = cv2.createCLAHE(clipLimit=5.0, tileGridSize=(8, 8))


class ImageUtils:
    """A Static-only Class to hold common image processing utilities & wrappers over OpenCV functions"""

    @staticmethod
    def save_img(path, final_marked):
        logger.info(f"Saving Image to '{path}'")
        cv2.imwrite(path, final_marked)

    @staticmethod
    def resize_util(img, u_width, u_height=None):
        if u_height is None:
            h, w = img.shape[:2]
            u_height = int(h * u_width / w)
        return cv2.resize(img, (int(u_width), int(u_height)))

    @staticmethod
    def resize_util_h(img, u_height, u_width=None):
        if u_width is None:
            h, w = img.shape[:2]
            u_width = int(w * u_height / h)
        return cv2.resize(img, (int(u_width), int(u_height)))

    @staticmethod
    def grab_contours(cnts):
        # source: imutils package

        # if the length the contours tuple returned by cv2.findContours
        # is '2' then we are using either OpenCV v2.4, v4-beta, or
        # v4-official
        if len(cnts) == 2:
            cnts = cnts[0]

        # if the length of the contours tuple is '3' then we are using
        # either OpenCV v3, v4-pre, or v4-alpha
        elif len(cnts) == 3:
            cnts = cnts[1]

        # otherwise OpenCV has changed their cv2.findContours return
        # signature yet again and I have no idea WTH is going on
        else:
            raise Exception(
                (
                    "Contours tuple must have length 2 or 3, "
                    "otherwise OpenCV changed their cv2.findContours return "
                    "signature yet again. Refer to OpenCV's documentation "
                    "in that case"
                )
            )

        # return the actual contours array
        return cnts

    @staticmethod
    def normalize_util(img, alpha=0, beta=255):
        return cv2.normalize(img, alpha, beta, norm_type=cv2.NORM_MINMAX)

    @staticmethod
    def auto_canny(image, sigma=0.93):
        # compute the median of the single channel pixel intensities
        v = np.median(image)

        # apply automatic Canny edge detection using the computed median
        lower = int(max(0, (1.0 - sigma) * v))
        upper = int(min(255, (1.0 + sigma) * v))
        edged = cv2.Canny(image, lower, upper)

        # return the edged image
        return edged

    @staticmethod
    def adjust_gamma(image, gamma=1.0):
        # build a lookup table mapping the pixel values [0, 255] to
        # their adjusted gamma values
        inv_gamma = 1.0 / gamma
        table = np.array(
            [((i / 255.0) ** inv_gamma) * 255 for i in np.arange(0, 256)]
        ).astype("uint8")

        # apply gamma correction using the lookup table
        return cv2.LUT(image, table)

    @staticmethod
    def four_point_transform(image, pts):
        # obtain a consistent order of the points and unpack them
        # individually
        rect = ImageUtils.order_points(pts)
        (tl, tr, br, bl) = rect

        # compute the width of the new image, which will be the
        width_a = np.sqrt(((br[0] - bl[0]) ** 2) + ((br[1] - bl[1]) ** 2))
        width_b = np.sqrt(((tr[0] - tl[0]) ** 2) + ((tr[1] - tl[1]) ** 2))

        max_width = max(int(width_a), int(width_b))
        # max_width = max(int(np.linalg.norm(br-bl)), int(np.linalg.norm(tr-tl)))

        # compute the height of the new image, which will be the
        height_a = np.sqrt(((tr[0] - br[0]) ** 2) + ((tr[1] - br[1]) ** 2))
        height_b = np.sqrt(((tl[0] - bl[0]) ** 2) + ((tl[1] - bl[1]) ** 2))
        max_height = max(int(height_a), int(height_b))
        # max_height = max(int(np.linalg.norm(tr-br)), int(np.linalg.norm(tl-br)))

        # now that we have the dimensions of the new image, construct
        # the set of destination points to obtain a "birds eye view",
        # (i.e. top-down view) of the image, again specifying points
        # in the top-left, top-right, bottom-right, and bottom-left
        # order
        dst = np.array(
            [
                [0, 0],
                [max_width - 1, 0],
                [max_width - 1, max_height - 1],
                [0, max_height - 1],
            ],
            dtype="float32",
        )

        transform_matrix = cv2.getPerspectiveTransform(rect, dst)
        warped = cv2.warpPerspective(image, transform_matrix, (max_width, max_height))

        # return the warped image
        return warped

    @staticmethod
    def order_points(pts):
        rect = np.zeros((4, 2), dtype="float32")

        # the top-left point will have the smallest sum, whereas
        # the bottom-right point will have the largest sum
        s = pts.sum(axis=1)
        rect[0] = pts[np.argmin(s)]
        rect[2] = pts[np.argmax(s)]
        diff = np.diff(pts, axis=1)
        rect[1] = pts[np.argmin(diff)]
        rect[3] = pts[np.argmax(diff)]

        # return the ordered coordinates
        return rect
