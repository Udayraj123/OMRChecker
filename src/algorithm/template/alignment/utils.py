import cv2
import numpy as np

from src.utils.image import ImageUtils
from src.utils.interaction import InteractionUtils


def show_displacement_overlay(
    block_gray_alignment_image, block_gray_image, shifted_block_image
):
    # ..
    overlay = ImageUtils.overlay_image(block_gray_alignment_image, block_gray_image)
    overlay_shifted = ImageUtils.overlay_image(
        block_gray_alignment_image, shifted_block_image
    )

    InteractionUtils.show(
        "Alignment For Field Block",
        ImageUtils.get_padded_hstack(
            [
                block_gray_alignment_image,
                block_gray_image,
                shifted_block_image,
                overlay,
                overlay_shifted,
            ]
        ),
    )


def hough_circles(gray):
    gray = cv2.medianBlur(gray, 5)
    rows = gray.shape[0]
    circles = cv2.HoughCircles(
        gray,
        cv2.HOUGH_GRADIENT,
        1,
        rows / 10,
        param1=100,
        param2=30,
        minRadius=1,
        maxRadius=30,
    )

    if circles is not None:
        circles = np.uint16(np.around(circles))
        # logger.info("circles", circles)
        for i in circles[0, :]:
            center = (i[0], i[1])
            # circle center
            cv2.circle(gray, center, 1, (0, 100, 100), 3)
            # circle outline
            radius = i[2]
            cv2.circle(gray, center, radius, (255, 0, 255), 3)
    return gray
