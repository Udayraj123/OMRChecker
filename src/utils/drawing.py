import cv2
import numpy as np

from src.utils.constants import CLR_BLACK, CLR_DARK_GRAY, CLR_GRAY, CLR_GREEN, TEXT_SIZE
from src.utils.image import ImageUtils
from src.utils.math import MathUtils


class DrawingUtils:
    @staticmethod
    def draw_matches(image, from_points, warped_image, to_points):
        horizontal_stack = ImageUtils.get_padded_hstack([image, warped_image])
        h, w = image.shape[:2]
        from_points = MathUtils.get_tuple_points(from_points)
        to_points = MathUtils.get_tuple_points(to_points)
        for from_point, to_point in zip(from_points, to_points):
            horizontal_stack = cv2.line(
                horizontal_stack,
                from_point,
                (w + to_point[0], to_point[1]),
                color=CLR_GREEN,
                thickness=3,
            )
        return horizontal_stack

    @staticmethod
    def draw_box_diagonal(
        image,
        position,
        position_diagonal,
        color=CLR_DARK_GRAY,
        border=3,
    ):
        cv2.rectangle(
            image,
            position,
            position_diagonal,
            color,
            border,
        )

    @staticmethod
    def draw_contour(
        image,
        contour,
        color=CLR_GREEN,
        thickness=2,
    ):
        assert None not in contour, "Invalid contour provided"
        cv2.drawContours(
            image,
            [np.intp(contour)],
            contourIdx=-1,
            color=color,
            thickness=thickness,
        )

    @staticmethod
    def draw_box(
        image,
        position,
        box_dimensions,
        color=None,
        style="BOX_HOLLOW",
        thickness_factor=1 / 12,
        border=3,
        centered=False,
    ):
        assert position is not None
        x, y = position
        box_w, box_h = box_dimensions

        position = (
            int(x + box_w * thickness_factor),
            int(y + box_h * thickness_factor),
        )
        position_diagonal = (
            int(x + box_w - box_w * thickness_factor),
            int(y + box_h - box_h * thickness_factor),
        )

        if centered:
            centered_position = [
                (3 * position[0] - position_diagonal[0]) // 2,
                (3 * position[1] - position_diagonal[1]) // 2,
            ]
            centered_diagonal = [
                (position[0] + position_diagonal[0]) // 2,
                (position[1] + position_diagonal[1]) // 2,
            ]
            position = centered_position
            position_diagonal = centered_diagonal

        if style == "BOX_HOLLOW":
            if color is None:
                color = CLR_GRAY
        elif style == "BOX_FILLED":
            if color is None:
                color = CLR_DARK_GRAY
            border = -1

        DrawingUtils.draw_box_diagonal(
            image,
            position,
            position_diagonal,
            color,
            border,
        )
        return position, position_diagonal

    @staticmethod
    def draw_arrows(
        image,
        start_points,
        end_points,
        color=CLR_GREEN,
        thickness=2,
        line_type=cv2.LINE_AA,
        tip_length=0.1,
    ):
        start_points = MathUtils.get_tuple_points(start_points)
        end_points = MathUtils.get_tuple_points(end_points)
        for start_point, end_point in zip(start_points, end_points):
            image = cv2.arrowedLine(
                image,
                start_point,
                end_point,
                color,
                thickness,
                line_type,
                tipLength=tip_length,
            )

        return image

    @staticmethod
    def draw_text(
        image,
        text_value,
        position,
        text_size=TEXT_SIZE,
        thickness=2,
        centered=False,
        color=CLR_BLACK,
        # available LineTypes: FILLED, LINE_4, LINE_8, LINE_AA
        line_type=cv2.LINE_AA,
        font_face=cv2.FONT_HERSHEY_SIMPLEX,
    ):
        if centered:
            assert not callable(
                position
            ), f"centered={centered} but position={position}"
            text_position = position
            position = lambda size_x, size_y: (
                text_position[0] - size_x // 2,
                text_position[1] + size_y // 2,
            )

        if callable(position):
            size_x, size_y = cv2.getTextSize(
                text_value,
                font_face,
                text_size,
                thickness,
            )[0]
            position = position(size_x, size_y)

        position = (int(position[0]), int(position[1]))
        cv2.putText(
            image,
            text_value,
            position,
            font_face,
            text_size,
            color,
            thickness,
            lineType=line_type,
        )

    @staticmethod
    def draw_symbol(image, symbol, position, position_diagonal, color=CLR_BLACK):
        center_position = lambda size_x, size_y: (
            (position[0] + position_diagonal[0] - size_x) // 2,
            (position[1] + position_diagonal[1] + size_y) // 2,
        )

        DrawingUtils.draw_text(image, symbol, center_position, color=color)

    @staticmethod
    def draw_line(image, start, end, color=CLR_BLACK, thickness=3):
        cv2.line(image, start, end, color, thickness)

    @staticmethod
    def draw_polygon(image, points, color=CLR_BLACK, thickness=1, closed=True):
        n = len(points)
        for i in range(n):
            if not closed and i == n - 1:
                continue
            DrawingUtils.draw_line(
                image, points[i % n], points[(i + 1) % n], color, thickness
            )

    @staticmethod
    def draw_group(
        image,
        start,
        bubble_dimensions,
        box_edge,
        color,
        thickness=3,
        thickness_factor=7 / 10,
    ):
        start_x, start_y = start
        box_w, box_h = bubble_dimensions
        if box_edge == "TOP":
            end_position = (start_x + int(box_w * thickness_factor), start_y)
            start = (start_x + int(box_w * (1 - thickness_factor)), start_y)
            DrawingUtils.draw_line(image, start, end_position, color, thickness)
        elif box_edge == "RIGHT":
            start = (start_x + box_w, start_y)
            end_position = (start_x, int(start_y + box_h * thickness_factor))
            start = (start_x, int(start_y + box_h * (1 - thickness_factor)))
            DrawingUtils.draw_line(image, start, end_position, color, thickness)
        elif box_edge == "BOTTOM":
            start = (start_x, start_y + box_h)
            end_position = (int(start_x + box_w * thickness_factor), start_y)
            start = (int(start_x + box_w * (1 - thickness_factor)), start_y)
            DrawingUtils.draw_line(image, start, end_position, color, thickness)
        elif box_edge == "LEFT":
            end_position = (start_x, int(start_y + box_h * thickness_factor))
            start = (start_x, int(start_y + box_h * (1 - thickness_factor)))
            DrawingUtils.draw_line(image, start, end_position, color, thickness)
