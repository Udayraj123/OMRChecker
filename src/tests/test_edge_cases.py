import os
from pathlib import Path

from src.tests.test_samples.sample2.boilerplate import (
    CONFIG_BOILERPLATE,
    TEMPLATE_BOILERPLATE,
)
from src.tests.utils import (
    generate_write_jsons_and_run,
    run_entry_point,
    setup_mocker_patches,
)

FROZEN_TIMESTAMP = "1970-01-01"
CURRENT_DIR = Path("src/tests")
BASE_SAMPLE_PATH = CURRENT_DIR.joinpath("test_samples", "sample2")


def run_sample(mocker, input_path):
    setup_mocker_patches(mocker)
    output_dir = os.path.join("outputs", input_path)
    run_entry_point(input_path, output_dir)
    # sample_outputs = extract_sample_outputs(output_dir)


write_jsons_and_run = generate_write_jsons_and_run(
    run_sample,
    sample_path=BASE_SAMPLE_PATH,
    template_boilerplate=TEMPLATE_BOILERPLATE,
    config_boilerplate=CONFIG_BOILERPLATE,
)


def test_config_low_dimensions(mocker):
    def modify_config(config):
        config["dimensions"]["processing_height"] = 1000
        config["dimensions"]["processing_width"] = 1000

    exception = write_jsons_and_run(mocker, modify_config=modify_config)

    assert str(exception) == "No Error"
