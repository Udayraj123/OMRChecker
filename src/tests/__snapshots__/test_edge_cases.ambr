# serializer version: 1
# name: test_config_low_dimensions_error_case
  'Error: No marker found in patch topLeftMarker'
# ---
# name: test_config_low_dimensions_safe_case
  None
# ---
